### PR TITLE
docs(agency): adding reference table for analysts to define agency, reference for pre-commit hooks

### LINKED ISSUE #1047: Docs request: what even is a transit agency and how can we answer questions that relate to their GTFS data?
A lot of times Caltrans staff will get asked various questions such as #984. When attempting to answer this question a lot of confusion can arise with how to quantify this answer due to our thorough dataset in the California Transit airtable. A quick how-to guide is needed to determine:

1. What organizations do we want to analyze depending on the research question?

There has been an effort (see the [Transit Provider Categorization](https://docs.google.com/document/d/1e0Z-aLKzNg7xnP0gSCCSWOuNPjTB8xN1G7vWVYQ8bYk/edit#heading=h.5taosqpsjhbq) doc) to classify organizations into various tracking, assistance or reporting categories. A shorter version of this should be documented such that analysts can quickly know what kind of filtering of organizations to use depending on the research ask.

2. How should an organization's services be aggregated when analyzing?

Typical research questions ask what seem to be fairly basic questions like #984. However, an organization can have multiple services that it manages and a service can be described in multiple GTFS datasets. Therefore, a situation can arise where it is unclear how the completeness of data should be counted. Is it when all services that an organization manages have a certain quality about them or when at least one, or a threshold? A clear plan of what aggregation strategy to use should be agreed upon and documented.

### PR DESCRIPTION
**Pull request by charlie-costanzo**
# Description

Creating agency reference table in the data services documentation so that analysts can easily interpret the use of the word agency in the research requests that they receive. Also added information for using and troubleshooting pre-commit hooks.

Misc changes: capitalized a lowercase page title, clarified the `python libraries` section

Resolves #1047 

New warehouse views for this ask generated in issue #1487 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation
- [ ] agencies.yml

## Files effected:
`warehouse / what_is_agency` page
* Added page with reference table 

`_toc.yml`
* Added `what_is_agency`

`contribute / submitting_changes`
* Added information for using and troubleshooting pre-commit hooks

`github/contribute_to_repos.md`
* Capitalize H1 (page title)

### CODE REVIEW

**Review by atvaccaro — APPROVED** [review]
lgtm but would like @lauriemerrell to review the agency definition when reasonable (low prio)

**Comment by charlie-costanzo** [comment]
> lgtm but would like @lauriemerrell to review the agency definition when reasonable (low prio)

@atvaccaro - @lauriemerrell and I have gone through all of the contents on this doc in the process of getting the new transit database table made for the PR, but I'd love any other thoughts that she might have since it's been some time now

File: docs/warehouse/what_is_agency.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
...
+| -------- | -------- |
+| **GTFS Datasets** | For both GTFS Static and GTFS Real-Time data, when trying to analyze GTFS datasets it is easiest to think of `agency` as "unique feed publisher", with the exception of feed `calitp_itp_id` == `200` as it is a regional reporter that publishes duplicates of other feeds that we also consume.<br/><br/>To identify unique feed publishers:<ul><li>Exclude `calitp_itp_id`== `200`</li><li>Deduplicate feeds</li></ul> |
+| **GTFS-Provider-Service Relationships** | In the warehouse, this is the relationship between `organizations` and the `services` they manage. <br/><br/>This is not an exhaustive list of all services managed by providers, only those that we are targeting to get into GTFS reporting.<br/><br/>Each record defines an organization and one of it's services. For the most part, each service is managed by a single organization with a small number of exceptions (e.g. Solano Express, which is jointly managed by Solano and Napa).<br/><br/>**Relevant data**:<br/>Table: `cal-itp-data-infra.mart_transit_database.dim_provider_gtfs_data`<ul><li>Column: `organization_name`</li><li>Column: `service_name`</li><br/>
```
Comment (charlie-costanzo) [inline docs/warehouse/what_is_agency.md:9]: Add a comment that says something like 'a question that uses this definition might look like this:" with examples

Comment (charlie-costanzo) [reply docs/warehouse/what_is_agency.md:9]: add explicit explanation of which definitions you used at the beginning of a notebook?

Comment (charlie-costanzo) [reply docs/warehouse/what_is_agency.md:9]: add: an 'agency' can be both organizations and service depending on context

Comment (charlie-costanzo) [reply docs/warehouse/what_is_agency.md:9]: is agency usually organization? should we suggest that?

**Comment by natam1** [comment]
@charlie-costanzo Added a new section to the agency docs, let me know what you think!

**Pull request merged**